python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -51,6 +52,25 @@
 log = logging.getLogger(__name__)
 
 
+def _lookupDataset(datasetType, registry, quantumDataId, collections) -> list[DatasetRef]:
```

Comment:
This function is generic enough, that I could see living in `lsst.utils` and importing from there.

Reply:
I think I'd prefer to leave it here and not advertise, because I'd like to get this fixed in pipe_base before it happens again.  I've created [DM-73286](https://jira.lsstcorp.org/browse/DM-37286) for that.